datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1554,12 +1580,16 @@ public ZkBackedLiveInstanceListProvider() {
     private List<String> getLiveInstanceNames(List<String> nodes) {
       List<String> liveInstances = new ArrayList<>();
       for (String n : nodes) {
-        String hostname = _zkclient.ensureReadData(KeyBuilder.liveInstance(_cluster, n));
-        if (hostname == null) {
-          // hostname can be null if a node dies immediately after reading all live instances
-          LOG.error("Node {} is dead. Likely cause it dies after reading list of nodes.", n);
-        } else {
-          liveInstances.add(formatZkInstance(hostname, n));
+        try {
+          String hostname = _zkclient.ensureReadData(KeyBuilder.liveInstance(_cluster, n));
+          if (hostname == null) {
+            // hostname can be null if a node dies immediately after reading all live instances
+            LOG.error("Node {} is dead. It died after the list of nodes were read.", n);
```

Comment:
Can you please reword this? Node dying is a bit confusing to read. The actual event might be helpful as in.. instance's session expired or something like that. Same for the exception below.